docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,576 @@
+that allows configuring permissions for the list of principals.
+
+**Principal** is an entity that can be authenticated.<br>
+**Permission** is access granted to the principal.  
+
+Each Kuma service has a generated certificate with URIs that identify service name and tags.
+For example "backend" service in "us-east" zone of version v1 can have the following list of URIs:
+* spiffe://mesh-1/backend
+* kuma://kuma.io/zone/us-east
+* kuma://version/v1
+
+If we want to grant any permissions to these instances we can create the following Network RBAC:
+
+```yaml
+action: ALLOW
+policies:
+  "backend-us-east-v1-allow-all":
+    permissions:
+      - any: true
+    principals:
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "spiffe://mesh-1/backend"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - authenticated:
+                principal_name:
+                  exact: "kuma://version/v1"
+```
+
+In order request to be allowed, at least 1 policy should be matched. 
+
+#### Converting MeshTrafficPermission to Envoy NetworkRBAC
+
+1. Concatenate "from" arrays from all matched MeshTrafficPermissions 
+
+Input
+```yaml
+type: MeshTrafficPermission
+name: global
+spec:
+  targetRef:
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+        action: DENY
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          kuma.io/zone: us-east
+      default:
+        action: ALLOW
+---
+type: MeshTrafficPermission
+name: backend
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  from:
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          env: dev
+      default:
+        action: DENY
+```
+
+Concatenated "from" array
+```yaml
+from:
+  - targetRef:
+      kind: Mesh
+    default:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    default:
+      action: ALLOW
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        env: dev
+    default:
+      action: DENY
+```
+
+2. Create a full rule-based view (with negations) 
+
+Regular [rule-based view](005-policy-matching.md#rule-based-view) doesn't work well for NetworkRBAC generation,
+because rule-based view relies on the order, but NetworkRBAC doesn't. 
+
+In this example, if a request matches "rule 2" this means it doesn't match "rule 1":
+```yaml
+rules:
+  - targetRef: # rule 1
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: dev
+    default:
+      action: DENY
+  - targetRef: # rule 2
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    default:
+      action: ALLOW
+```
+
+So "rule 2" implicitly matches requests from zone "us-east" that are **not** from env "dev".
```

Comment:
Why, isn't this something we decide? Isn't this a problem in general with `MeshSubset` and "specificity" of tags?

What's the problem with making the order matter here?

If `env: dev` is first, then `env: dev` requests are allowed and `!env: dev` requests are allowed.
If `env: dev` is second, then the rule is essentially ignored?

Reply:
Discussed offline, can we resolve this?